channel/signedstate.go

```diff
@@ -1,17 +1,60 @@
 package channel
 
 import (
+	"errors"
+
 	"github.com/statechannels/go-nitro/channel/state"
 )
 
-type SignedState struct {
-	State state.VariablePart
-	Sigs  map[uint]state.Signature // keyed by participant index
+type signedState struct {
+	state state.State
+	sigs  map[uint]state.Signature // keyed by participant index
+}
+
+// newSignedState initializes a SignedState struct for the given state.
+// The signedState returned will have no signatures.
+func newSignedState(s state.State) signedState {
+	return signedState{s, make(map[uint]state.Signature, len(s.Participants))}
+}
+
+// addSignature adds a participant's signature for the state.
+// An error is thrown if the signature is invalid.
+func (ss signedState) addSignature(sig state.Signature) error {
+	signer, err := ss.state.RecoverSigner(sig)
+	if err != nil {
+		return err
```

Comment:
This is potentially a good candidate for error wrapping, like we do here: 

https://github.com/statechannels/go-nitro/blob/b70f16695f8baf807e5338b132b55b861b5278d9/channel/state/state.go#L101

Reply:
https://go.dev/blog/go1.13-errors

Reply:
Good idea! Updated in 5992af9